docs/understand/architecture.md

```diff
@@ -57,6 +72,35 @@ The credential scheme, along with the request signing logic are compatible with
 
 Currently, the auth service manages its own database of users and credentials and does not use IAM in any way. 
 
+### Hooks Engine
+
+The hooks engine enables CI/CD for data by triggering user defined [Actions](../setup/hooks.md) that will run during commit/merge. 
+
 ### Frontend UI
 
 The UI layer is a simple browser-based client that uses the OpenAPI server. It allows management, exploration and data access to repositories, branches, commits and objects in the system.
+
+## lakeFS Clients
+
+There are several ways of interacting with lakeFS. Many data applications have different use-cases for lakeFS and 
+we try to support as many as possible.
+
+### OpenAPI Generated SDKs
+
+OpenAPI specification can be used to generate lakeFS clients for many programming languages.
+For example, the Python [lakefs-client](https://pypi.org/project/lakefs-client/) or the [Java client](https://search.maven.org/artifact/io.lakefs/api-client).
+
+### lakectl
+
+[lakectl](../reference/commands.md) is a CLI tool that enables lakeFS operations using the lakeFS API from your preferred terminal.
+
+### Spark Metadata Client
+
+The lakeFS [Spark Metadata Client](../reference/spark-client.md) makes it easy to perform
+operations related to lakeFS metadata. Examples include [garbage collection](../reference/garbage-collection.md) or [exporting data from lakeFS](../reference/export.md).
+
+### lakeFS Hadoop FileSystem
+
+While it is possible to interact with lakeFS using Hadoop's S3AFIleSystem, doing so may introduce performance problems.
+Using [lakeFSFileSystem](../integrations/spark.md#access-lakefs-using-the-lakefs-specific-hadoop-filesystem) increases Spark ETL jobs performance by executing the metadata operations on the lakeFS server,
```

Comment:
```suggestion
Using [lakeFSFileSystem](../integrations/spark.md#access-lakefs-using-the-lakefs-specific-hadoop-filesystem) preserve Spark ETL jobs performance by executing the metadata operations on the lakeFS server,
```

Reply:
`Increases` is better 😈

Reply:
but is it true? :)

Reply:
Compared to using lakeFS without lakeFSFilesystem? For high volumes it is true

Reply:
>Compared to using lakeFS without lakeFSFilesystem
that is correct. I was thinking about the comparison to using Spark in general without lakeFS